pkg/retagger/job_pattern.go

```diff
@@ -22,18 +27,65 @@ type PatternJob struct {
 	Options JobOptions
 }
 
+type backoffTransport struct {
+	Transport http.RoundTripper
+	logger    micrologger.Logger
+}
+
+func (t *backoffTransport) RoundTrip(request *http.Request) (*http.Response, error) {
+	var err error
+	var resp *http.Response
+
+	{
+		o := func() error {
+			var respErr error
+			resp, respErr = t.Transport.RoundTrip(request)
+			// Internal error, return nil to prevent retry
+			if respErr != nil {
+				err = respErr
+				return nil
+			}
+			// Rate limited
+			if resp.StatusCode == 429 {
+				return microerror.New("rate limited")
```

Comment:
Need an error type to match here?

Reply:
Good call. This was a placeholder. I might have forgotten about it. Fixing now.